src/Bedrock.Framework.Experimental/Transports/WebSockets/WebSocketWriteFrame.cs

```diff
@@ -0,0 +1,40 @@
+﻿using System;
+using System.Buffers;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Bedrock.Framework.Experimental.Transports.WebSockets
+{
+    /// <summary>
+    /// A WebSocket frame to write as output.
+    /// </summary>
+    public class WebSocketWriteFrame
```

Comment:
I'd love to see some examples of using this. What does the header look like and are there invalid headers?

Reply:
I'm not following you there. There are tests for the writer coming, if that's the question.

Reply:
I'm not following you there. If you mean tests, those are forthcoming.

Reply:
This API is a tad bit lower level than I expected but it's probably fine. It means the caller needs to provide a masking key (presumably that also determines server vs client  while the corefx API has an `isServer` bool)

Reply:
Ah, I gotcha. Yeah, when you said write raw frames and a consumer would worry about the details, I was thinking you meant that kind of detail also. To me, that makes the most sense at this layer, which I was not thinking was going to handle the actual protocol details at all (like when it's valid to send certain opcodes, how to deal with close frames, who is allowed to mask, etc.) But, I'm certainly open to ideas here.